test/unit/filters/NormalFilterTest.cpp

```diff
@@ -251,4 +251,64 @@ TEST(NormalFilterTest, RampPlane2)
     }
 }
 
+TEST(NormalFilterTest, RadiusSearch)
+{
+    using namespace Dimension;
+
+    PointTable table;
+    table.layout()->registerDims({Id::X, Id::Y, Id::Z});
+
+    BufferReader reader;
+    NormalFilter filter;
+    Options filterOps;
+    filterOps.add("radius", 2.0);
+    filter.setInput(reader);
+    filter.setOptions(filterOps);
+    filter.prepare(table);
+
+    PointViewPtr view(new PointView(table));
+    view->setField(Id::X, 0, 0);
+    view->setField(Id::X, 1, 0);
+    view->setField(Id::X, 2, 1);
+    view->setField(Id::X, 3, 1);
+    view->setField(Id::X, 4, 9);
+    view->setField(Id::X, 5, 9);
+    view->setField(Id::X, 6, 10);
+    view->setField(Id::X, 7, 10);
+    view->setField(Id::Y, 0, 0);
+    view->setField(Id::Y, 1, 1);
+    view->setField(Id::Y, 2, 0);
+    view->setField(Id::Y, 3, 1);
+    view->setField(Id::Y, 4, 9);
+    view->setField(Id::Y, 5, 10);
+    view->setField(Id::Y, 6, 9);
+    view->setField(Id::Y, 7, 10);
+    view->setField(Id::Z, 0, 0);
+    view->setField(Id::Z, 1, 1);
+    view->setField(Id::Z, 2, 0);
+    view->setField(Id::Z, 3, 1);
+    view->setField(Id::Z, 4, 9);
+    view->setField(Id::Z, 5, 10);
+    view->setField(Id::Z, 6, 9);
+    view->setField(Id::Z, 7, 10);
+    reader.addView(view);
+
+    PointViewSet viewSet = filter.execute(table);
+    PointViewPtr outView = *viewSet.begin();
+
+    Dimension::Id nx = table.layout()->findDim("NormalX");
+    Dimension::Id ny = table.layout()->findDim("NormalY");
+    Dimension::Id nz = table.layout()->findDim("NormalZ");
+    Dimension::Id c = table.layout()->findDim("Curvature");
+
+    double expected = std::sqrt(2.0) / 2.0;
+    for (auto const& p : *outView)
```

Comment:
Not a fan of `auto const` here. What is `p`? I have no idea. What's gained by `auto`? I know people do this all over, but I still think it's bad practice unless the alternative is worse.